src/Microsoft.ApplicationInsights.AspNetCore/RequestTrackingTelemetryModule.cs

```diff
@@ -74,16 +74,14 @@ public void Initialize(TelemetryConfiguration configuration)
                         }
 
                         this.diagnosticListeners.Add(new HostingDiagnosticListener(
+                            configuration,
                             this.telemetryClient,
                             this.applicationIdProvider,
                             this.CollectionOptions.InjectResponseHeaders,
                             this.CollectionOptions.TrackExceptions,
                             this.CollectionOptions.EnableW3CDistributedTracing,
                             enableNewDiagnosticEvents));
 
-                        this.diagnosticListeners.Add
```

Comment:
do we want to keep the list? there is no reason for it,  and no future reason. Can we just keep HostingListener as instance field and subscribe it?